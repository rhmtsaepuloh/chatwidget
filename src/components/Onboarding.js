import React from "react";
import { hexToRgbA, defaultText } from "../utils/ui";
import Button from "./Button";
import Loading from "./Loading";

import "../styles/Onboarding.scss";
import LogoPowered from "./LogoPowered";

const Onboarding = ({ dataClient, loading, onRegister, content }) => {
  const {
    logo,
    primaryColor,
    secondaryColor,
    buttonColor,
    primaryTextColor,
    secondaryTextColor,
    buttonTextColor,
  } = dataClient;
  const { contentOpening } = content;
  const white = "#FFFFFF"
  const rgb = (opacity) => hexToRgbA(primaryColor, opacity);
  const rgb2 = (opacity) => hexToRgbA(white, opacity);

  return (
    <div className='wrapper-onboarding'>
      <div
        className='main-background flex flex-col justify-between'
        style={{ background: rgb(1) }}>
        <div className='flex items-center justify-between pt-4 pl-4'>
          <p
            className='text-base font-bold'
            style={{ color: primaryTextColor }}>
            {defaultText(contentOpening, "textHeader", "ChatAja Widget")}
          </p>
        </div>
        <div
          className='flex items-center justify-between lg:h-64 lg:w-64 h-40 w-40 rounded-full mx-auto'
          style={{ background: rgb2(0.09) }}>
          <div
            className='flex items-center justify-between lg:h-48 lg:w-48 h-32 w-32 rounded-full mx-auto'
            style={{ background: rgb2(0.07) }}>
            <div
              className='flex items-center justify-between lg:h-32 lg:w-32 h-20 w-20 rounded-full mx-auto'
              style={{ background: rgb2(0.05) }}>
              <img
                alt='logo'
                src={logo}
                className='h-12 w-12 rounded-full mx-auto'
              />
            </div>
          </div>
        </div>
        <div className='text-center px-10'>
          <p
            className='text-white text-base font-bold'
            style={{ color: primaryTextColor }}>
            {defaultText(contentOpening, "textJudul", "Selamat Datang!")}
          </p>
          <p className='text-white text-sm' style={{ color: primaryTextColor }}>
            {defaultText(
              contentOpening,
              "textCaption",
              "Kamu dapat menanyakan apa saja kepada customer service hanya dengan chat..."
            )}
          </p>
        </div>
        <Button
          buttonText={defaultText(contentOpening, "textTombol", "Mulai Chat")}
          onClick={onRegister}
          icon
          background={buttonColor}
          buttonColor={buttonTextColor}
        />
      </div>
      <div
        className='footer-background flex items-center justify-center mt-8 inset-x-0 fixed'
        style={{
          bottom: "0px",
          padding: "10px 0px",
          background: secondaryColor,
        }}>
        <LogoPowered color={secondaryTextColor} />
        <p
          className='text-xs text-center box-border ml-1'
          style={{ color: secondaryTextColor }}>
          Powered by ChatAja
        </p>
      </div>
      <Loading display={loading} />
    </div>
  );
};

export default Onboarding;
