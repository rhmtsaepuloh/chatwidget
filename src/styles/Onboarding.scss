.wrapper-onboarding {
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  @media (max-height: 450px) {
    padding-bottom: 80px;
    background: #fff;
  }
  .main-background {
    height: 400px;
    border-top-left-radius: 20px;
    border-top-right-radius: 20px;
    z-index: 1;
    @media (max-height: 541px) {
      margin: 0px;
      height: 400px;
    }
    @media (min-height: 541px) and (max-height: 604px) {
      margin: 0px;
      height: 480px;
    }
    @media (min-height: 604px) and (max-height: 704px) {
      margin: 0px;
      height: 540px;
    }
    @media (min-height: 704px) and (max-height: 804px) {
      margin: 0px;
      height: 620px;
    }
    @media (min-height: 804px) and (max-height: 1048px) {
      margin: 0px;
      height: 720px;
    }
    @media (min-height: 1048px) {
      margin: 0px;
      height: 1000px;
    }
    .rating-circle {
      margin: 40px auto 0px;
      @media (max-height: 541px) {
        margin: 0px auto 0px;
      }
      @media (min-height: 541px) and (max-height: 604px) {
        margin: 50px auto 0px;
      }
      @media (min-height: 604px) and (max-height: 704px) {
        margin: 60px auto 0px;
      }
      @media (min-height: 704px) and (max-height: 804px) {
        margin: 110px auto 0px;
      }
      @media (min-height: 804px) and (max-height: 1048px) {
        margin: 120px auto 0px;
      }
      @media (min-height: 1048px) {
        margin: 300px auto 0px;
      }
    }
    .thanks {
      @media (min-height: 804px) and (max-height: 850px) {
        position: absolute;
        bottom: 180px;
      }
    }
  }
  .button-position {
    margin-top: -20px;
  }
  .footer-background {
    background: #fff;
    border-radius: 0px 0px 20px 20px;
    margin: auto;
    @media (max-height: 500px) {
      border-radius: 0px;
    }
    @media (max-width: 768px) {
      width: 100%;
    }
    @media (min-width: 768px) {
      width: 378px;
    }
  }
}
