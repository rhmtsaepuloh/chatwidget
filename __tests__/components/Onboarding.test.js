/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import Onboarding from "../../src/components/Onboarding";

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe("render correctly onboarding components", () => {
  test("renders", () => {
    const dataClient = {
      hexColor: "#CA1234",
      logo:
        "https://chatwidget-chataja.s3.ap-southeast-1.amazonaws.com/0fcedfb7-1861-42c0-9515-13603ce62901.png",
    };
    const content = {
      contentOpening: {
        textHeader: "ChatAja Widget",
        textJudul: "ChatAja",
        textCaption:
          "Kamu dapat menanyakan apa saja kepada customer service hanya dengan chat...",
        textTombol: "Mulai Chat",
      },
    };
    const wrapper = shallow(
      <Onboarding dataClient={dataClient} content={content} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
