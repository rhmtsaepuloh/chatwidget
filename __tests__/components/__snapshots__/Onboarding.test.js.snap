// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly onboarding components renders 1`] = `ShallowWrapper {}`;
